src/lib/pipeline/FetchUnit.cc

```diff
@@ -176,8 +176,16 @@ void FetchUnit::tick() {
       }
     } else if (loopBufferState_ == LoopBufferState::WAITING &&
                pc_ == loopBoundaryAddress_) {
-      // Once set loopBoundaryAddress_ is fetched, start to fill loop buffer
-      loopBufferState_ = LoopBufferState::FILLING;
+      // Now that the loopBoundaryAddress_ has been set, start filling Loop
+      // Buffer if the branch predictor tells us to reenter the detected loop
+      if (macroOp[0]->isBranch() && !macroOp[0]->getBranchPrediction().taken) {
```

Comment:
Replace first part with "loopBoundaryAddress_ has been fetched whilst loop buffer is waiting,"  for clarity and consistency